ArdusatSDK.cpp

```diff
@@ -453,266 +257,1212 @@ const char * valuesToCSV(const char *sensorName, unsigned long timestamp, int nu
+ * @ingroup rgblight
+ *
+ * @param   sensor_id Specifies which RGB sensor to use: TCS34725 or ISL29125
+ *     - SENSORID_TCS34725
+ *     - SENSORID_ISL29125
+ *
+ * Example Usage:
+ * @code
+ *     RGBLight rgb = RGBLight(SENSORID_ISL29125); // Instantiate sensor object
+ *     rgb.begin();                                // Initialize sensor
+ *     Serial.println(rgb.readToJSON());           // Read and print values in JSON
+ * @endcode
+ *****************************************************************************/
+RGBLight::RGBLight(sensor_id_t sensor_id) {
+  RGBLight::sensorId = sensor_id;
+  RGBLight::islIntensity = CFG1_10KLUX;
+  RGBLight::tcsIt = TCS34725_INTEGRATIONTIME_154MS;
+  RGBLight::tcsGain = TCS34725_GAIN_1X;
+  RGBLight::header.unit = DATA_UNIT_LUX;
+  RGBLight::initialized = false;
+}
+
+RGBLight::~RGBLight() {
+}
+
+/**
+ * @brief   Initializes the sensor with basic configurations
+ * @ingroup rgblight
+ *
+ * @retval true  Successfully initialized
+ * @retval false Failed to initialize
+ */
+boolean RGBLight::begin(void) {
+  return RGBLight::begin(RGBLight::islIntensity, RGBLight::tcsIt, RGBLight::tcsGain);
+}
+
+/**
+ * @brief   Initializes the sensor with advanced configurations
+ * @ingroup rgblight
+ *
+ * @param islIntensity Advanced configuration for ISL29125 intensity
+ *     - CFG1_375LUX
+ *     - CFG1_10KLUX (Default)
+ *
+ * @retval true  Successfully initialized
+ * @retval false Failed to initialize
+ */
+boolean RGBLight::begin(uint8_t islIntensity) {
+  if (RGBLight::sensorId == SENSORID_ISL29125)
+    return RGBLight::begin(islIntensity, RGBLight::tcsIt, RGBLight::tcsGain);
+  else
+    return RGBLight::begin();
+}
+
+/**
+ * @brief   Initializes the sensor with advanced configurations
+ * @ingroup rgblight
+ *
+ * @param tcsIt Advanced configuration for TCS34725 integration time
+ *     - TCS34725_INTEGRATIONTIME_2_4MS
+ *     - TCS34725_INTEGRATIONTIME_24MS
+ *     - TCS34725_INTEGRATIONTIME_50MS
+ *     - TCS34725_INTEGRATIONTIME_101MS
+ *     - TCS34725_INTEGRATIONTIME_154MS (Default)
+ *     - TCS34725_INTEGRATIONTIME_700MS
+ *
+ * @param tcsGain Advanced configuration for TCS34725 gain
+ *     - TCS34725_GAIN_1X (Default)
+ *     - TCS34725_GAIN_4X
+ *     - TCS34725_GAIN_16X
+ *     - TCS34725_GAIN_60X
+ *
+ * @retval true  Successfully initialized
+ * @retval false Failed to initialize
+ */
+boolean RGBLight::begin(tcs34725IntegrationTime_t tcsIt, tcs34725Gain_t tcsGain) {
+  if (RGBLight::sensorId == SENSORID_TCS34725)
+    return RGBLight::begin(RGBLight::islIntensity, tcsIt, tcsGain);
+  else
+    return RGBLight::begin();
+}
+
+/**
+ * Private function to initialize RGBLight sensor
+ */
+boolean RGBLight::begin(uint8_t islIntensity, tcs34725IntegrationTime_t tcsIt, tcs34725Gain_t tcsGain) {
+  RGBLight::islIntensity = islIntensity;
+  RGBLight::tcsIt = tcsIt;
+  RGBLight::tcsGain = tcsGain;
+
+  if (RGBLight::sensorId == SENSORID_ISL29125)
```

Comment:
I _really_ dislike using the "two line" `if` syntax in C/C++ - way too easy to add a line and mess things up. Use explicit brackets here.